### DISCUSSION TITLE
First time contributing

### DISCUSSION THREAD

**Discussion by JohnKara** [beetbox/beets#4246] [Q&A] [answered]
Hello everyone, my name is John and I am currently studying Computer Science. 

As an assignment for the Software Engineering course, I have to contribute to an open source project and I found beets really interesting. 

Since this is my first time contributing to an open source project, It would be great if someone could give me some guidelines and perhaps something to work on. 

I have already installed beets on my Windows machine and have taken some time to see how it works. I noticed that the output of cmd is a little bit messy and also I could not import all my tracks with “beet import path” command. Nevertheless, I managed to tag all the tracks I have in my library. 

I'm looking forward to your suggestions!

**Comment by sampsyo** [comment, marked as answer]
Great to meet you, @JohnKara!

So we can try to guide you in the right direction, can you give us a little more background on what you're interested in (and capable of) doing? The best option, if you're comfortable with Python, might be to fix one bug and see how that goes—given time, maybe you can address other issues, but starting small is a good way to get familiar with the codebase. For example, you could start by looking through the "bitesize" label we sometimes use on issues that are fairly self-contained:
https://github.com/beetbox/beets/issues?q=is%3Aopen+is%3Aissue+label%3Abitesize

**Reply by JohnKara** [reply]
Thank you for your quick response @sampsyo!

I've used python in various fields such as machine learning, computational intelligence, cryptography, pattern recognition and some other small projects. 

I will try to fix a bug or implement a feature from these issues. 

Is there anywhere information about the structure of the project?

**Reply by sampsyo** [reply]
Great! Take a look in the issue tracker and see what appeals to you.

You might be interested in reading over the ["For Developers"](https://beets.readthedocs.io/en/stable/dev/index.html) section of the docs to get started. The [Concepts](https://github.com/beetbox/beets/wiki/Concepts) wiki page could also be helpful.
